include/ensmallen_bits/sa/sa_impl.hpp

```diff
@@ -219,12 +220,12 @@ inline void SA<CoolingScheduleType>::MoveControl(const size_t nMoves,
                                                  MatType& accept,
                                                  MatType& moveSize)
 {
-  MatType target;
-  target.copy_size(accept);
+  MatType target(accept.n_rows, accept.n_cols);
   target.fill(0.44);
-  moveSize = arma::log(moveSize);
+
+  moveSize = log(moveSize);
   moveSize += gain * (accept / (double) nMoves - target);
-  moveSize = arma::exp(moveSize);
+  moveSize = exp(moveSize);
 
   // To avoid the use of element-wise arma::min(), which is only available in
   // Armadillo after v3.930, we use a for loop here instead.
```

Comment:
Wow, this is ancient!  You could replace this part now too, and it'll be a lot faster on the GPU.

Reply:
Handled in dcbdcdc4.